### PR TITLE
Bug 1118444 - Check against all navigator.languages values for possible supported locales.

### PR DESCRIPTION
**Pull request by Standard8**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[Standard8 Mark Banner (Standard8)](https://github.com/Standard8) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=22a28192cd2a)

**Comment by try-server-hook** [comment]
[Standard8 Mark Banner (Standard8)](https://github.com/Standard8) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=86d5fa2264c1)

**Comment by Standard8** [comment]
I just pushed a minor update to avoid an else after an early return.

File: apps/sharedtest/test/unit/l10n/lib/context/langnego_test.js [hunk 1]
```diff
@@ -44,3 +44,20 @@ describe('Language negotiation with arguments', function() {
...
 });
+
+describe('Language negotiation with arguments and unsupported', function() {
+  var ctx;
+
+  beforeEach(function(done) {
+    ctx = new Context();
+    ctx.once(done);
+    ctx.registerLocales('en-US', ['pl']);
+    ctx.requestLocales('pl-PT', 'pl');
```
Comment (stasm) [inline apps/sharedtest/test/unit/l10n/lib/context/langnego_test.js]: Can you use something else than `pl-PT` here? Maybe `de` or `fr` which I think are often used in other l10n.js tests.

File: shared/js/l10n.js [hunk 2]
```diff
@@ -1348,12 +1348,21 @@
   // Getting ready
 
   function negotiate(available, requested, defaultLocale) {
-    if (available.indexOf(requested[0]) === -1 ||
-        requested[0] === defaultLocale) {
+    var supportedLocale;
+    // Find the first locale in the requested list that is supported.
+    for (var i = 0; i < requested.length; ++i) {
```
Comment (stasm) [inline shared/js/l10n.js]: Is there any difference between `++i` and `i++` here?  It seems that in other parts of the code we use post-incrementing for loop counters.

Comment (Standard8) [inline shared/js/l10n.js]: Probably not, in a past life I used to do that a lot, and haven't done recently, don't know why I did this time ;-) I'll change it.

File: apps/sharedtest/test/unit/l10n/lib/context/langnego_test.js (hunk 1 above)
Comment (Standard8) [inline apps/sharedtest/test/unit/l10n/lib/context/langnego_test.js]: pl seems to be the one used in this directory, and in the fixtures sub-directory, there's only pl.properties and en-US.properties.

I also took a look, and de/fr aren't so convenient as there accept languages start off with the locale they are actually for.

**Pull request merged**